.github/workflows/swagger.yml

```
@@ -1,22 +1,37 @@
 name: Open API Specification and Documentation
-on: 
+on:
   push:
-    paths-ignore:
-      - '.github/**'
+    branches:
+      - "master"
+    paths:
+      - "handlers/**"
   pull_request:
-      paths-ignore:
-      - '.github/**'
+    paths:
+      - "handlers/**"
 jobs:
   swaggerci:
+    if: github.repository == 'meshery/meshery'
     name: swagger-docs
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.GH_ACCESS_TOKEN }}
+          ref: "master"
       - name: Setup go-swagger
         uses: minchao/setup-go-swagger@v1
         with:
           version: v0.26.1
       - name: swagger-spec
         run: swagger generate spec -o ./helpers/swagger.yaml --scan-models
       - name: swagger-docs
-        run: swagger generate spec -o ./docs/_data/swagger.yml --scan-models
+        run: swagger generate spec -o ./docs/_data/swagger.yml --scan-models && swagger flatten ./docs/_data/swagger.yml -o ./docs/_data/swagger.yml --with-expand --format=yaml
+      - name: Commit changes
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          file_pattern: docs helpers/
+          commit_user_name: l5io
+          commit_user_email: ci@layer5.io
+          commit_author: ${{ github.actor }} <${{ github.actor }}@users.noreply.github.com>
+          commit_options: "--signoff"
+          commit_message: "[Swagger-Docs] Updated the OpenAPI files"

```

```suggestion
          commit_message: "[Docs] Updated swagger docs for REST API"

```
Hmm.. it would be cool if this commit msg told about the docs that it was updating. :thinking: 
We're updating reference for REST API, right?

---

right!